src/consensus/encode.rs

```diff
@@ -393,14 +388,22 @@ pub struct CheckedData(pub Vec<u8>);
 // Primitive types
 macro_rules! impl_int_encodable{
     ($ty:ident, $meth_dec:ident, $meth_enc:ident) => (
-        impl<D: Decoder> Decodable<D> for $ty {
+        impl Decodable for $ty {
             #[inline]
-            fn consensus_decode(d: &mut D) -> Result<$ty, self::Error> { d.$meth_dec().map($ty::from_le) }
+            fn consensus_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
+                ReadExt::$meth_dec(&mut d).map($ty::from_le)
+            }
         }
 
-        impl<S: Encoder> Encodable<S> for $ty {
+        impl Encodable for $ty {
             #[inline]
-            fn consensus_encode(&self, s: &mut S) -> Result<(), self::Error> { s.$meth_enc(self.to_le()) }
+            fn consensus_encode<S: WriteExt>(
```

Comment:
Wait, so now we have WriteExt, but only for a few types, and most just take io::Write? I mean that mostly prevents people from making the mistake I made of trying to implement ReadExt, but not fully. Can we just drop WriteExt (maybe in a future PR)?

Reply:
Everything that implements `io::Write` implements `WriteExt`. No, we cannot "just drop WriteExt"; this trait is needed to encoded Bitcoin types.